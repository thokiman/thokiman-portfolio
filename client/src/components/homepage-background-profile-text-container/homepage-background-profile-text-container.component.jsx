import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { selectIsSideBarHidden } from "../../redux/header/header.selectors";
import {
  HomepageActivityText,
  HomepageBackgroundProfileText,
  HomepageNameText,
  HomepageProfileTextContainer,
  HomepageQuoteText,
} from "./homepage-background-profile-text-container.styles";

const HomePageProfileTextContainer = ({
  shortDescription,
  quote,
  isSideBarHidden,
}) => {
  return (
    <HomepageProfileTextContainer $issidebarhidden={isSideBarHidden}>
      <HomepageBackgroundProfileText>
        <HomepageNameText>Thomas Santosa</HomepageNameText>
        <HomepageActivityText>{shortDescription}</HomepageActivityText>
        <HomepageQuoteText>{quote}</HomepageQuoteText>
      </HomepageBackgroundProfileText>
    </HomepageProfileTextContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isSideBarHidden: selectIsSideBarHidden,
});

export default connect(mapStateToProps)(HomePageProfileTextContainer);
