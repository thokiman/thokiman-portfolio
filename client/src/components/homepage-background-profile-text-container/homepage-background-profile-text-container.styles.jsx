import styled, { css } from "styled-components";
const SideBarActive = css`
  display: none;
`;
const getSideBarActive = (props) => {
  if (props.$issidebarhidden) return SideBarActive;
  return null;
};
export const HomepageProfileTextContainer = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  @media only screen and (max-width: 800px) {
    background-color: #3f4249;
    ${getSideBarActive}
  }
`;

export const HomepageBackgroundProfileText = styled.div`
  width: 100%;
  height: 75%;
  padding: 0 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const HomepageNameText = styled.div`
  text-align: center;
  font-kerning: auto;
  font-family: angilla;
  letter-spacing: 0.005em;
  font-size: 800%;
  width: fit-content;
  height: fit-content;
  word-spacing: -30px;
  padding-right: 10px;
  margin-bottom: 20px;
  @media only screen and (max-width: 800px) {
    padding-top: 40px;
    font-size: 17vw;
    word-spacing: -18px;
    text-align: center;
  }
`;
const HomepageTextStyles = css`
  color: #9096ae;
  font-family: baskerville-old-face;
  font-kerning: auto;
  padding-right: 10px;
`;
export const HomepageActivityText = styled.div`
  ${HomepageTextStyles}
  font-size: 200%;
  letter-spacing: 0.1em;
  line-height: 1.1em;
  width: 100%;
  height: fit-content;
  margin-bottom: 10px;
  text-align: center;
  @media only screen and (max-width: 800px) {
    text-align: left;
    width: 80%;
    font-size: 7vw;
    letter-spacing: 0.05em;
  }
`;

export const HomepageQuoteText = styled.div`
  ${HomepageTextStyles}
  font-size: 140%;
  letter-spacing: 0.1em;
  line-height: 1.15em;
  width: 90%;
  height: fit-content;
  text-align: right;
  @media only screen and (max-width: 800px) {
    text-align: right;
    width: 80%;
    font-size: 4.5vw;
    letter-spacing: 0.05em;
  }
`;
HomepageProfileTextContainer.displayName = "HomepageProfileTextContainer";
HomepageBackgroundProfileText.displayName = "HomepageBackgroundProfileText";
HomepageNameText.displayName = "HomepageNameText";
HomepageActivityText.displayName = "HomepageActivityText";
HomepageQuoteText.displayName = "HomepageQuoteText";
