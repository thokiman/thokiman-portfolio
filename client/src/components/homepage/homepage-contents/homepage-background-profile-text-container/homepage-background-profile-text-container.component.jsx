import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectIsSideBarActive } from '../../../../redux/header/header.selectors';
import {
  HomepageActivityText,
  HomepageBackgroundProfileText,
  HomepageNameText,
  HomepageProfileTextContainer,
  HomepageQuoteText,
} from './homepage-background-profile-text-container.styles';

export const HomePageProfileTextContainer = ({
  shortDescription,
  quote,
  isSideBarHidden: isSideBarActive,
}) => {
  return (
    <HomepageProfileTextContainer $issidebaractive={isSideBarActive}>
      <HomepageBackgroundProfileText>
        <HomepageNameText>Thomas Santosa</HomepageNameText>
        <HomepageActivityText>{shortDescription}</HomepageActivityText>
        <HomepageQuoteText>{quote}</HomepageQuoteText>
      </HomepageBackgroundProfileText>
    </HomepageProfileTextContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isSideBarActive: selectIsSideBarActive,
});

export default connect(mapStateToProps)(HomePageProfileTextContainer);
