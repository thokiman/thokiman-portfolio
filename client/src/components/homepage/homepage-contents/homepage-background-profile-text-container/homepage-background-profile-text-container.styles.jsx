import styled, { css } from 'styled-components';

const backgroundProfileText = '#000000';
const lightGrayColor = '#9096ae';
const maxWidthDesktop = 1366;

const HomepageIsNotActived = css`
  display: none;
`;
const getSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return HomepageIsNotActived;
  return null;
};
export const HomepageProfileTextContainer = styled.div`
  height: 45%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  @media only screen and (max-width: ${maxWidthDesktop}px) {
    background-color: ${backgroundProfileText};
    ${getSideBarActive}
    height: max-content;
  }
`;

export const HomepageBackgroundProfileText = styled.div`
  width: 100%;
  height: 75%;
  padding: 0 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const HomepageNameText = styled.div`
  text-align: center;
  font-kerning: auto;
  font-family: angilla;
  letter-spacing: 0.005em;
  font-size: 800%;
  width: fit-content;
  height: fit-content;
  word-spacing: -30px;
  padding-right: 10px;
  margin-bottom: 20px;
  @media only screen and (max-width: ${maxWidthDesktop}px) {
    padding-top: 40px;
    font-size: 17vw;
    word-spacing: -18px;
    text-align: center;
  }
`;
const HomepageTextStyles = css`
  color: ${lightGrayColor};
  font-family: baskerville-old-face;
  font-kerning: auto;
  padding-right: 10px;
`;
export const HomepageActivityText = styled.div`
  ${HomepageTextStyles}
  font-size: 200%;
  letter-spacing: 0.1em;
  line-height: 1.1em;
  width: 100%;
  height: fit-content;
  margin-bottom: 10px;
  text-align: center;
  @media only screen and (max-width: ${maxWidthDesktop}px) {
    text-align: left;
    width: 80%;
    font-size: 7vw;
    letter-spacing: 0.05em;
  }
`;

export const HomepageQuoteText = styled.div`
  ${HomepageTextStyles}
  font-size: 140%;
  letter-spacing: 0.1em;
  line-height: 1.15em;
  width: 90%;
  height: fit-content;
  text-align: right;
  @media only screen and (max-width: ${maxWidthDesktop}px) {
    text-align: right;
    width: 80%;
    font-size: 4.5vw;
    letter-spacing: 0.05em;
  }
`;
HomepageProfileTextContainer.displayName = 'HomepageProfileTextContainer';
HomepageBackgroundProfileText.displayName = 'HomepageBackgroundProfileText';
HomepageNameText.displayName = 'HomepageNameText';
HomepageActivityText.displayName = 'HomepageActivityText';
HomepageQuoteText.displayName = 'HomepageQuoteText';
