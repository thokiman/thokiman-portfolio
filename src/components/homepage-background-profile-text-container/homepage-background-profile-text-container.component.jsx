import React from "react";

import {
  HomepageActivityText,
  HomepageBackgroundProfileText,
  HomepageNameText,
  HomepageProfileTextContainer,
  HomepageQuoteText,
} from "./homepage-background-profile-text-container.styles";

const HomePageProfileTextContainer = ({ shortDescription, quote }) => {
  return (
    <HomepageProfileTextContainer>
      <HomepageBackgroundProfileText>
        <HomepageNameText>Thomas Santosa</HomepageNameText>
        <HomepageActivityText>{shortDescription}</HomepageActivityText>
        <HomepageQuoteText>{quote}</HomepageQuoteText>
      </HomepageBackgroundProfileText>
    </HomepageProfileTextContainer>
  );
};

export default HomePageProfileTextContainer;
