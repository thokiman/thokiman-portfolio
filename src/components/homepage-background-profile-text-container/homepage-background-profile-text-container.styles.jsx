import styled, { css } from "styled-components";

export const HomepageProfileTextContainer = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  @media only screen and (max-width: 800px) {
    background-color: #3f4249;
  }
`;

export const HomepageBackgroundProfileText = styled.div`
  width: 100%;
  height: 75%;
  padding: 0 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const HomepageNameText = styled.div`
  text-align: center;
  font-kerning: auto;
  font-family: angilla;
  letter-spacing: 0.005em;
  font-size: 800%;
  width: fit-content;
  height: fit-content;
  word-spacing: -30px;
  padding-right: 10px;
  margin-bottom: 20px;
`;
const HomepageTextStyles = css`
  color: #9096ae;
  font-family: baskerville-old-face;
  font-kerning: auto;
  padding-right: 10px;
`;
export const HomepageActivityText = styled.div`
  ${HomepageTextStyles}
  font-size: 200%;
  letter-spacing: 0.1em;
  line-height: 1.1em;
  width: 100%;
  height: fit-content;
  margin-bottom: 10px;
  text-align: center;
`;

export const HomepageQuoteText = styled.div`
  ${HomepageTextStyles}
  font-size: 140%;
  letter-spacing: 0.1em;
  line-height: 1.15em;
  width: 90%;
  height: fit-content;
  text-align: right;
`;
HomepageProfileTextContainer.displayName = "HomepageProfileTextContainer";
HomepageBackgroundProfileText.displayName = "HomepageBackgroundProfileText";
HomepageNameText.displayName = "HomepageNameText";
HomepageActivityText.displayName = "HomepageActivityText";
HomepageQuoteText.displayName = "HomepageQuoteText";
