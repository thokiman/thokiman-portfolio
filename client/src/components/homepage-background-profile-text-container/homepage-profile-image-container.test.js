import React from "react";
import { shallow } from "enzyme";
import { HomePageProfileTextContainer } from "./homepage-background-profile-text-container.component";

it("should render HomePageProfileTextContainer component", () => {
  expect(shallow(<HomePageProfileTextContainer />)).toMatchSnapshot();
});
