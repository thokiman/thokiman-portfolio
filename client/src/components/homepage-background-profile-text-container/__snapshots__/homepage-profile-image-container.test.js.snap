// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePageProfileTextContainer component 1`] = `ShallowWrapper {}`;
